R/summarize_ts.R

```diff
@@ -1,42 +1,57 @@
 #' Summarize a timeseries
 #' 
 #' @param var_src character vector of dataset types to summarize.
-#' @param site_name character vector of the same length as var_src specifying the 
-#'   site or sites (e.g., "nwis_24309857") where each timeseries dataset should 
-#'   be sought.
+#' @param site_name character vector of the same length as var_src specifying
+#'   the site or sites (e.g., "nwis_24309857") where each timeseries dataset
+#'   should be sought.
+#' @param out a list of one or more outputs to include in the summary dataframe,
+#'   in addition to the var_src and site_name and id columns.
 #' @param ... other args passed to \code{\link[sbtools]{session_check_reauth}}
 #' @import dplyr
 #' @export
 #' @examples
 #' \dontrun{
-#' summarize_ts("doobs_nwis", "nwis_01021050")
+#' summarize_ts(rep(c("doobs_nwis", "wtr_nwis"), each=4), 
+#'   rep(c("nwis_01021050","nwis_01036390","nwis_01073389","nwis_notasite"), times=2))
 #' }
-summarize_ts <- function(var_src, site_name, ...) {
-  # collect the inputs and determine whether each row describes an existing ts
-  ts_summary <- 
-    data.frame(var_src=var_src, site=site_name, stringsAsFactors=FALSE) %>%
-    mutate(id=locate_ts(var_src=var_src, site_name=site_name))
+summarize_ts <- function(var_src, site_name, out=c("start_date","end_date","num_rows","num_complete"), ...) {
   
-  # create a data.frame of summary metrics, one per row
+  # check inputs & session
+  out <- match.arg(out, several.ok=TRUE)
```

Comment:
i think it rarely makes sense to run `match.arg` outside of a function. try this:
```
outfun <- function(out=c("start_date","end_date","num_rows","num_complete")) {
  return(match.arg(out, several.ok=TRUE))
}
outfun()
```